### PR TITLE
reside-63: workflow for use within a package

### PR DESCRIPTION
**Pull request by richfitz**
This PR will add new functionality for making it easier to use the translator from a package.  Similar to how we handled the cache in hintr, we don't want to have to pass a translation object and language everywhere around the package.  Instead, we'd like to just be able to write

```r
t_("string")
```

and have it translate appropriately.  This is one approach, including an example package that implements hello-world-as-a-service

### CODE REVIEW

**Comment by richfitz** [comment]
Fixes #2 - thanks, @maelle

File: vignettes/traduire.Rmd [hunk 1]
```diff
@@ -61,3 +64,100 @@ translator$t("interpolate", list(what = "i18next", how = "easy"),
...
+
+
+
+To do this, we allow packages (or other simiar code) to "register" a translator, like
+
+```r
+traduire::translator_register(resources)
+```
+
+where `name` is the key representing your object (more on this later) and `...` are arguments to `traduire::i18n`
```
Comment (r-ash) [inline vignettes/traduire.Rmd]: Is this line out of date? Where are `name` and `...` used? Similar what does `resources` on line 86 refer to?

**Review by r-ash — APPROVED** [review]
This looks really good - just what we need. The `preroute` and `postserialize` stuff will make this super easy in `hintr`.

Thinking about `naomi` it would be nice to see an example using a file where we want to translate some text within it.

I guess we would have something like in our json for model options
```
{
  "label": "GENERAL",
  "description": "GENERAL_DESCRIPTION",
  "controlGroups": [
    {
      "label": "AREA_SCOPE",
      "controls": [
        {
          "name": "area_scope",
          "type": "multiselect",
          "options": 
            <area_scope_options>,
          "value": <area_scope_default>,
          "required": true
        }
      ]
    },
...
```
Then our translation file something like
```
{
  "gb": {
    "translation": {
      "GENERAL": "General",
      "GENERAL_DESCRIPTION": "Select general model options:",
      "AREA_SCOPE": "Area scope"
    }
  },
  "fr": {
    "translation": {
      "GENERAL": "Générales",
      "GENERAL_DESCRIPTION": "Sélectionner les options générales du modèle",
      "AREA_SCOPE": "portée de la zone"
    }
  },
...
```
but with more accurate French translation. Then calling `t_` on the big load of json would be able to translate those keys within the text?

**Comment by richfitz** [comment]
I've logged the replacement use-case as [RESIDE-64](https://vimc.myjetbrains.com/youtrack/issue/RESIDE-64) and have an idea to implement that fairly cleanly. I'll open a new PR later today with an implementation

File: vignettes/traduire.Rmd (hunk 1 above)
Comment (richfitz) [reply vignettes/traduire.Rmd]: thanks, updated

**Pull request merged**